src/backend/executor/nodeShareInputScan.c

```diff
@@ -666,6 +717,29 @@ static int retry_write(int fd, char *buf, int wsize)
 	return 0;
 }
 
+
+
+/*
+ * mk_tmp_file_name
+ *
+ * Called by reader or writer to make the unique tmp file name.
+ */
+void mk_tmp_file_name(char* p, int size, int share_id, const char* name)
```

Comment:
If the name of the temp file for writer follows the naming convention of its ready and done notification fifo, I think we can eliminate this function.

It is good enough to have a single function like joint_tmp_file_name which generate name for the temp file. This function can then be used at both the creation and drop stage of the temp file.